webui/react/src/components/UPlotChart.tsx

```diff
@@ -55,7 +57,33 @@ const UPlotChart: React.FC<Props> = ({ data, focusIndex, options }: Props) => {
 
     const optionsExtended = uPlot.assign(
       {
-        cursor: { drag: { dist: 5, uni: 10, x: true, y: true } },
+        cursor: {
+          bind: {
+            dblclick: (_uPlot: uPlot, _target: EventTarget, handler: (e: Event) => void) => {
+              return (e: Event) => {
+                isZoomed.current = false;
+                handler(e);
+              };
+            },
+            mousedown: (_uPlot: uPlot, _target: EventTarget, handler: (e: Event) => void) => {
+              return (e: MouseEvent) => {
+                mousePosition.current = [ e.clientX, e.clientY ];
+                handler(e);
+              };
+            },
+            mouseup: (_uPlot: uPlot, _target: EventTarget, handler: (e: Event) => void) => {
+              return (e: MouseEvent) => {
+                const distX = e.clientX-mousePosition.current[0];
+                const distY = e.clientY-mousePosition.current[1];
+                if (Math.sqrt(distX**2 + distY**2) > 5) {
+                  isZoomed.current = true;
+                }
```

Comment:
**chore:** under `src/utils/chart.ts` there is a `distance` function that we can use here.